### ISSUE TITLE
Indenter rearranges my line of code.

### ISSUE THREAD

**Issue by p45cal** [rubberduck-vba/Rubberduck#4431]
Before Indenting
`Sub sdf4()
  ctxt = "SELECT `Sequence ID` as Header1, '' as Header2 , Sum(`No Of Try`) AS Header3, '' as Header4 ,'' as Header5 ,'' as Header6 ,'' as Header7 ,'' as Header8 ,'' as Header9 ,'' as Header10 FROM `'" & sheetName & "$'` GROUP BY `KIT NO`, DEVICE, `Sequence ID` HAVING (`KIT NO`='" & KitNo & "')"
End Sub
`

and after:
`Sub sdf4()
ctxt = "SELECT `Sequence ID` as Header1, '' as Header2 , Sum(`No Of Try`) AS Header3, '' as Header4 ,'' as Header5 ,'' as Header6 ,'' as Header7 ,'' as Header8 ,'' as Header9 ,'' as Header10 FROM `'" & sheetName & " & KitNo & " ')"` GROUP BY `KIT NO`, DEVICE, `Sequence ID` HAVING (`KIT NO`='" & KitNo & "
End Sub
`
Version:
Version 2.2.0.3516
OS: Microsoft Windows NT 10.0.17134.0, x64
Host Product: Microsoft Office 2016 x86
Host Version: 16.0.9126.2282
Host Executable: EXCEL.EXE

TraceLog, Indenter config and screenshot files attached.
[Indenter settings.txt](https://github.com/rubberduck-vba/Rubberduck/files/2473803/Indenter.settings.txt)
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/2473794/RubberduckLog.txt)
![capture](https://user-images.githubusercontent.com/44096047/46881194-a834dc00-ce42-11e8-97e5-14746d3c2519.JPG)

**Comment by retailcoder** [comment]
Confirmed repro with 2.2.0.3971; a whole chunk of the string is getting dropped, and weird non-ANSI characters are being appended inside the string literal:

![image](https://user-images.githubusercontent.com/5751684/46882105-82481100-ce1b-11e8-9741-2504a4bb5612.png)

**Comment by retailcoder** [comment]
Interestingly, splitting the logical line into multiple concatenated lines fixes it:

```vb
    ctxt = "SELECT `Sequence ID` as Header1, '' as Header2 , Sum(`No Of Try`) AS Header3, '' as Header4 ,'' as Header5 ,'' as Header6 ,'' as Header7 ,'' as Header8 ,'' as Header9 ,'' as Header10 " & _
    "FROM `'" & sheetName & "$'` " & _
    "GROUP BY `KIT NO`, DEVICE, `Sequence ID` HAVING (`KIT NO`='" & KitNo & "')"
```

correctly indents to:

```vb
    ctxt = "SELECT `Sequence ID` as Header1, '' as Header2 , Sum(`No Of Try`) AS Header3, '' as Header4 ,'' as Header5 ,'' as Header6 ,'' as Header7 ,'' as Header8 ,'' as Header9 ,'' as Header10 " & _
           "FROM `'" & sheetName & " & _` " & _
           "GROUP BY `KIT NO`, DEVICE, `Sequence ID` HAVING (`KIT NO`='" & KitNo & "')"
```

**Comment by p45cal** [comment]
Not quite, the $ symbol after sheetName is absent.

Oh, you've just upated with that.

**Comment by retailcoder** [comment]
Thanks for reporting the bug!

That said if this connects to a database then you should probably be using a parameterized query instead of concatenating the `KitNo` criteria into the string. Not that it fixes anything, but still.
Also, the filtering would be more efficient in a `WHERE` clause, since `HAVING` filter is applied *after* the `GROUP BY` - filtering *before* would make the grouping process fewer rows.

**Comment by p45cal** [comment]
> you should probably be using a parameterized query instead of concatenating the `KitNo` criteria into the string. Not that it fixes anything, but still.
> Also, the filtering would be more efficient in a `WHERE` clause, since `HAVING` filter is applied _after_ the `GROUP BY` - filtering _before_ would make the grouping process fewer rows.

Yes, this is one of my early forays in SQL. I've never used a parameterized query. Putting that aside for a minute, which order would you have SELECT, WHERE and GROUP BY in such a statement?
(It creates a ADODB.Recordset from an Excel worksheet in another workbook.)
Off topic, I know.

**Comment by retailcoder** [comment]
@p45cal exactly in the order you listed them ;-)

**Comment by p45cal** [comment]
`  ctxt = "SELECT `Sequence ID` as Header1, '' as Header2 , Sum(`No Of Try`) AS Header3, '' as Header4 ,'' as Header5 ,'' as Header6 ,'' as Header7 ,'' as Header8 ,'' as Header9 ,'' as Header10 FROM `'" & sheetName & "$'` WHERE (`KIT NO`='" & KitNo & "') GROUP BY `KIT NO`, DEVICE, `Sequence ID`"
`
was 1% _slower_!  (4 columns, 7k rows of Excel Table in a sheet)
[We'll be able to delete these off-topic msgs won't we?]
